### PR TITLE
Build nuget package and make it part of Build.cmd

### PR DESCRIPTION
**Pull request by basoundr**
We are creating the Nuget package to give us the ability to insert ourselves to the VS Setup Authoring, which currently external APIs as Nuget packages

tagging @dotnet/roslyn-analysis for review.

### CODE REVIEW

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    <!-- Since VS is the only consumer, now, the version is set to 1.0.0-->
+    <version>1.0.0</version>
+    <authors>Managed Languages</authors>
+    <licenseUrl>http://go.microsoft.com/fwlink/?LinkId=394369</licenseUrl>
+    <projectUrl>https://github.com/dotnet/roslyn-project-system</projectUrl>
+    <releaseNotes>Preview of Roslyn based Project System</releaseNotes>
+    <tags>Roslyn ProjectSystem PS CSharp VisualBasic VisualStudio</tags>
+  </metadata>
+  <files>
+    <file src="Microsoft.VisualStudio.AppDesigner.dll" target="lib\net46" />
```
Comment (basoundr) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: 'pkgdef' for  'Microsoft.VisualStudio.AppDesigner' and 'Microsoft.VisualStudio.Editors' will be added once we have the appropriate pkgdef info removed from the existing VS components which add the pkgdef today

File: build/Targets/Version.targets [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <AssemblyVersion>1.0.0</AssemblyVersion>
+    <IsReleaseVersion>false</IsReleaseVersion>
+  </PropertyGroup>
+
+  <Choose>
+    <When Condition="$(BuildNumber) == ''">
+      <PropertyGroup>
+        <NuGetVersionSuffix>rc2</NuGetVersionSuffix>
```
Comment (davkean) [inline build/Targets/Version.targets]: rc2?

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec (hunk 1 above)
Comment (davkean) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: Where's the manifest? Not sure I follow the reason for not adding the Editors/AppDesigner pkgdef? How is this different from the dll's themselves which are also duplicated in VS?

File: build/Targets/Version.targets [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <AssemblyVersion>1.0.0</AssemblyVersion>
```
Comment (davkean) [inline build/Targets/Version.targets]: This isn't correct. AppDesigner/Editors needs to be 15.0.0.0.

How is this file different from VSL.Versions.targets? Should we remove the later?

File: build/build.proj [hunk 4]
```diff
@@ -36,6 +37,14 @@
              Targets="Build"
              Properties="$(CommonMSBuildGlobalProperties)"
              />
+
```
Comment (davkean) [inline build/build.proj]: What does this have to do with RestorePackages? This is building a nuget package - the reverse. :) 

I would put this in its own target.

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuget.proj [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (davkean) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuget.proj]: I think we should call this package something else, this isn't the nuget package for the project system, this the nuget package for *insertion* into VS.

File: build/Targets/Version.targets [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      <PropertyGroup>
+        <NuGetVersion>$(AssemblyVersion)</NuGetVersion>
+        <NuGetVersionType>Release</NuGetVersionType>
+      </PropertyGroup>
+    </When>
+    
+    <Otherwise>
+      <PropertyGroup>
+        <NuGetVersion>$(AssemblyVersion)-$(NuGetVersionSuffix)</NuGetVersion>
+        <NuGetVersionType>PerBuildPreRelease</NuGetVersionType>
```
Comment (davkean) [inline build/Targets/Version.targets]: What does "PerBuildPreRelease" mean?

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  <metadata>
+    <id>Microsoft.VisualStudio.ProjectSystem</id>
+    <summary>CoreXT package for Microsoft Project System for C# and VB </summary>
+    <description>
+      This Project system is a re-write of the C# and VB project system in Managed Language
+    </description>  
+    <language>en-US</language>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <!-- Since VS is the only consumer, now, the version is set to 1.0.0-->
+    <version>1.0.0</version>
```
Comment (davkean) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: Isn't this version just overwritten? ie 1.0.0 is meaningless because you pass the version above, right?

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec [hunk 8]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd">
+  <metadata>
+    <id>Microsoft.VisualStudio.ProjectSystem</id>
+    <summary>CoreXT package for Microsoft Project System for C# and VB </summary>
```
Comment (davkean) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: Change CoreXT to something that isn't a internal name, like "Visual Studio Insertion Package or something something"

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuget.proj (hunk 5 above)
Comment (basoundr) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuget.proj]: I am modifying the name to `Microsoft.VisualStudio.ProjectSystem.VSInsertionNuget.proj`

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec (hunk 8 above)
Comment (basoundr) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: sure

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec (hunk 7 above)
Comment (basoundr) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: This value is not overwritten and `1.0.0` is the only value

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec (hunk 1 above)
Comment (basoundr) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: Currently we dont copy pkgdef to output. I thought there was some reason behind it. If not, then I can modify the project to copy it to output and include it in the Nuget.

btw, what manifest are you talking about? I dont see any manifest output.

File: build/build.proj (hunk 4 above)
Comment (basoundr) [inline build/build.proj]: This target is not under RestorePackages. The github diff gave you the illusion :)

But I will move it to a separate target

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec (hunk 7 above)
Comment (basoundr) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: I tried removing the version from the nuspec but Nuget tool complains missing version information. Guess it is just required. Cannot do much about it

**Comment by basoundr** [comment]
@davkean I have addressed your comments. Please take a look when you can

**Comment by davkean** [comment]
:+1:

File: build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec (hunk 1 above)
Comment (davkean) [inline build/Nuget/Microsoft.VisualStudio.ProjectSystem.nuspec]: It's in the VSIX, we should copy to output given you're not using the VSIX. Basically, anything in the VSIX needs to be installed.

**Comment by davkean** [comment]
Sorry, I thumbs up without seeing some of your comments, can you follow up with my new comments with a new PR?

**Comment by basoundr** [comment]
@davkean sure. what other new comments you have?

**Comment by davkean** [comment]
> Where's the manifest? Not sure I follow the reason for not adding the Editors/AppDesigner pkgdef? How is this different from the dll's themselves which are also duplicated in VS?
> It's in the VSIX, we should copy to output given you're not using the VSIX. Basically, anything in the VSIX needs to be installed.
